src/to-be-visible.js

```diff
@@ -0,0 +1,40 @@
+import { matcherHint } from 'jest-matcher-utils';
+import { mergeAll } from 'ramda';
+
+import { checkReactElement, printElement } from './utils';
+
+function isStyleVisible(element) {
+  const style = element.props.style || {};
+  const { opacity = 1 } = Array.isArray(style) ? mergeAll(style) : style;
+  return opacity !== 0;
+}
+
+function isAttributeVisible(element) {
+  return element.type !== 'Modal' || !element.props.visible === false;
+}
+
+function isElementVisible(element) {
+  return (
+    isStyleVisible(element) &&
+    (!element.parent || isStyleVisible(element.parent)) &&
```

Comment:
I may be misinterpreting the code, but this appears to only check one ancestor. I.e. if the subject element is a grandchild, this will not verify the grandparent's visibility. Could we leverage recursion and achieve's [`jest-dom`'s](https://github.com/testing-library/jest-dom#tobevisible) "its parent element is also visible (and so on up to the top of the DOM tree)" ability? 

```js
function isElementVisible(element) {
  return (
    isStyleVisible(element) &&
    isAttributeVisible(element) &&
    (!element.parent || isElementVisible(element.parent))
  );
}
```

Reply:
Right, the idea is to traverse all element's ancestors in order to check that both style and attribute are visible. In fact, the recursion to achieve this is already covered by the following line:

https://github.com/fluiddot/jest-native/blob/feat/to-be-visible/src/to-be-visible.js#L21

I think I incorrectly added this line as it's not really necessary, I'll remove it as soon as possible, thanks for catching this 🙇 .

Reply:
I fixed this issue in https://github.com/testing-library/jest-native/commit/96c6ac525e1e547540fbeb9b27bc8056d214cb95.

Reply:
I fixed this issue in https://github.com/testing-library/jest-native/commit/5dd7ffe7abebf82a5fe810c24da02baf1973aa21.